### ISSUE TITLE
[no-unused-vars] False positive on args used in type predicate

### ISSUE THREAD

**Issue by zozoens31** [typescript-eslint/typescript-eslint#1299]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/<rule>": ["no-unused-vars"]
  }
}
```

```TS
interface Model {
  contentId: number
  contentName: string
  valueId: number
  valueName: string
}

function keepNames(model: Model): string[] {
  return Object.entries(model).
    filter(([key, value]): value is string => key.endsWith('Name')).
    map(([key, value]): string => value)
}
```

**Expected Result**
No error

**Actual Result**
`error  eslint:@typescript-eslint/no-unused-vars  'value' is defined but never used`

**Additional Info**
I can fix this by using a `argsIgnorePattern` and use it to change the name of `value`, but it seems wrong, since I'm actually using the `value` (the actual pattern I use is `^unused[A-Z]` which isn't something you expect to see used somewhere).

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.7.0` |
| `@typescript-eslint/parser`        | `2.7.0` |
| `TypeScript`                       | `3.7.2` |
| `ESLint`                           | `6.6.0` |
| `node`                             | `12.10.0` |
| `npm`                              | `6.11.3` |

**Comment by bradzacher** [comment]
please use our rule `no-unused-vars-experimental` instead.
